pyalgostrategypool/stochastic_crossover.py

```diff
@@ -0,0 +1,195 @@
+import talib
+from pyalgotrading.constants import *
+from pyalgotrading.strategy.strategy_base import StrategyBase
+
+import logging
+logger = logging.getLogger('strategy')
+
+
+class StochasticCrossover(StrategyBase):
+
+    def __init__(self, *args, **kwargs):
+        """
+        Accept and sanitize all your parameters here
+        setup the variables you will need here
+        if you are running the strategy for multiple days, then this method will be called only once at the start of the strategy
+        """
+        super().__init__(*args, **kwargs)
+
+        # Stochastic parameters
+        self.fastk_period = self.strategy_parameters['FASTK_PERIOD']
+        self.slowk_period = self.strategy_parameters['SLOWK_PERIOD']
+        self.slowd_period = self.strategy_parameters['SLOWD_PERIOD']
+
+        # Sanity
+        assert (0 < self.fastk_period == int(self.fastk_period)), f"Strategy parameter FASTK_PERIOD should be a positive integer. Received: {self.fastk_period}"
+        assert (0 < self.slowk_period == int(self.slowk_period)), f"Strategy parameter SLOWK_PERIOD should be a positive integer. Received: {self.slowk_period}"
+        assert (0 < self.slowd_period == int(self.slowd_period)), f"Strategy parameter SLOWD_PERIOD should be a positive integer. Received: {self.slowd_period}"
+
+        # Variables
+        self.main_order = None
+
+    def initialize(self):
+        """
+        If you are running the strategy for multiple days, then this method will be called once at the start of every day
+        use this to initialize and re-initialize your variables
+        """
+        self.main_order = {}
+
+    @staticmethod
+    def name():
+        """
+        Name of your strategy
+        """
+        return 'Stochastic Crossover'
+
+    @staticmethod
+    def versions_supported():
+        """
+        Strategy should always support the latest engine version
+        Current version is 3.3.0
+        """
+        return AlgoBullsEngineVersion.VERSION_3_3_0
+
+    def get_decision(self, instrument):
+        """
+        This method calculates the crossover using the hist data of the instrument along with the required indicator
+        """
+
+        # Get OHLC historical data for the instrument
+        hist_data = self.get_historical_data(instrument)
+
+        # Calculate the Stochastic values
+        slowk, slowd = talib.STOCH(hist_data['high'], hist_data['low'], hist_data['close'], fastk_period=self.fastk_period,
+                                   slowk_period=self.slowk_period, slowk_matype=0, slowd_period=self.slowd_period, slowd_matype=0)
+
+        # Get the crossover value
+        stochastic_crossover_value = self.utils.crossover(slowk, slowd)
+
+        if stochastic_crossover_value == 1:
+            action = 'BUY'
+        elif stochastic_crossover_value == -1:
+            action = 'SELL'
+        else:
+            action = None
```

Comment:
lets print the crossover and action here also, like other strategies

Reply:
Print not added because crossover gets printed anyways and entry/exit is present in the order transaction type, please let me know if this is correct.